Wrappers/Python/cil/optimisation/functions/L1Norm.py

```diff
@@ -37,6 +38,8 @@ def soft_shrinkage(x, tau, out=None):
     --------
     the value of the soft-shrinkage operator at x: DataContainer.
     """
+    if  id(x)==id(out):
+        raise InPlaceError
```

Comment:
Why not fix this as proposed here
https://github.com/TomographicImaging/CIL/blob/0118e61652d82d884662002fb40f46c0cb7c3812/Wrappers/Python/cil/optimisation/functions/L1Norm.py#L24-L70

Reply:
thanks @paskino - In this PR we are raising the errors and catching them in the unit tests. Future issues and PRs will fix the in-place errors

Reply:
For reference I leave the issue about it https://github.com/TomographicImaging/CIL/issues/1662